pkg/sanity/controller.go

```diff
@@ -587,7 +587,14 @@ var _ = DescribeSanity("Controller Service [Controller Server]", func(sc *TestCo
 					},
 				},
 			}
-			_, err := r.CreateVolume(context.Background(), vol2Req)
+			volume2, err := r.CreateVolume(context.Background(), vol2Req)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Clean up sourced volume first to prevent plugin from returning an
+			// in-use error.
+			By("cleaning up deleting the volume created from snapshot")
+			delVol2Req := MakeDeleteVolumeReq(sc, volume2.GetVolume().GetVolumeId())
+			_, err = r.DeleteVolume(context.Background(), delVol2Req)
```

Comment:
This looks like the cleanup code didn't delete volumes in the right order, correct?

I think we should fix that instead of adding custom cleanup code in the tests themselves. Suppose we add another check after creating the second volume and before deleting it: if that check then fails, test cleanup will fail like it does now.

Reply:
I haven't checked how volumes are tracked, but perhaps clean up with last-in-first-out order?

Reply:
We [currently always delete snapshots first, then volumes](https://github.com/kubernetes-csi/csi-test/blob/0ddc0e0dd7df52b42db71e48c890fd3c254add10/pkg/sanity/resources.go#L251-L252). Deletion in LIFO order sounds like the right approach to me, I think it should work. (For this particular test case, it means that we'd delete volume 2 first, then the snapshot, then the volume 1.)

I'll work on making the necessary updates to this PR.

Reply:
I made the discussed adjustments. Will comment some of the changes via Github comments to add more clarity.

Please have a look and let me know what you think.